src/WebJobs.Script.Scaling/AzureEnvironments.cs

```diff
@@ -0,0 +1,200 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Win32;
+
+namespace Microsoft.Azure.WebJobs.Script.Scaling
+{
+    public static class AzureEnvironments
+    {
+        private static string _runtimeSiteName;
+        private static string _hostName;
+        private static string _homeStampName;
+        private static string _currentStampName;
+        private static string _workerName;
+        private static string _storageConnectionString;
+        private static bool? _validateCertificates;
+
+        /// <summary>
+        /// Gets or sets runtime site name
+        /// </summary>
+        public static string SiteName
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_runtimeSiteName))
+                {
+                    var runtimeSiteName = Environment.GetEnvironmentVariable("WEBSITE_IIS_SITE_NAME").ToLowerInvariant();
+                    if (runtimeSiteName.StartsWith("~1"))
+                    {
+                        _runtimeSiteName = runtimeSiteName.Substring(2);
+                    }
+                    else
+                    {
+                        _runtimeSiteName = runtimeSiteName;
+                    }
+                }
+
+                return _runtimeSiteName;
+            }
+
+            set
+            {
+                _runtimeSiteName = value;
```

Comment:
This is simply reading env.  The writable part are all for testing purpose.   In real situation, no one should set it.    Since it is internal assembly (not for public), I don't put any internal visibility or any.   Keep in simple and make them all public.